File: Wirtualny_Swiat_Java/Projekt_PO_Java/src/model/Organisms/BarszczSosnowskiego.java
package model.Organisms;

import model.Organism;
import model.Plant;
import model.Point;
import model.World;

import java.awt.*;

public class BarszczSosnowskiego extends Plant {
    private final Color barszczColor = Color.RED;

    public BarszczSosnowskiego(World world, Point position){
        super(world, position, 10);
    }

    @Override
    public Color getColor(){
        return barszczColor;
    }

    @Override
    public String getNazwa(){
        return "BarszczSosnowskiego";
    }

    @Override
    public Organism makeNewOrganism(Point position){
        return new BarszczSosnowskiego(world, position);
    }

    @Override
    public void Action(){
        super.Action();

        Point currentPos = getPosition();

        for (int i = 1; i < 5; i++){
            Point target = new Point(currentPos.getX(), currentPos.getY());

            switch (i){
                case 1 -> target.setY(target.getY() - 1);
                case 2 -> target.setX(target.getX() + 1);
                case 3 -> target.setY(target.getY() + 1);
                case 4 -> target.setX(target.getX() - 1);
            }

            if (world.isCorrectPosition(target) && !world.isEmpty(target)){
                Organism other = world.getOrganismaAtPosition(target);

                if (other instanceof Cyber_Sheep){
                    continue;
                }

                if (other instanceof BarszczSosnowskiego){
                    continue;
                }

                if (other instanceof Human){
                    world.killHuman();
                    continue;
                }

                other.setAlive(false);
                Point otherPosition = other.getPosition();
                world.addEvent("Barszcz Sosnowskiego zabil " + other.getNazwa() + " na pozycji " + otherPosition.getX() + ", " + otherPosition.getY() + ")");
            }
        }
    }
}

File: WirtualnySwiatPython/Plants/BarszczSosnowskiego.py
from Animals.Human import Human
from plant import Plant
from config import RED, GORA, PRAWO, DOL, LEWO
from point import Point


class BarszczSosnowskiego(Plant):
    def __init__(self, world, position):
        super().__init__(world, position, power=10)
        self.color = RED

    def getColor(self):
        return self.color

    def getNazwa(self):
        return "BarszczSosnowskiego"

    def makeNewOrganism(self, position):
        return BarszczSosnowskiego(self.world, position)

    def Action(self):
        super().Action()

        current = self.getPosition()

        for i in range(4):
            x = current.GetX()
            y = current.GetY()

            if i == GORA:
                y -= 1
            elif i == PRAWO:
                x += 1
            elif i == DOL:
                y += 1
            elif i == LEWO:
                x -= 1

            target = Point(x, y)
            if (self.world.isCorrectPosition(target) and not self.world.isEmptyPosition(target)):
                from Animals.CyberSheep import CyberSheep
                other = self.world.getOrganismAtPosition(target)

                if isinstance(other, CyberSheep):
                    continue

                if isinstance(other, BarszczSosnowskiego):
                    continue

                if isinstance(other, Human):
                    self.world.killHuman()
                    continue

                other.setIsAlive(False)
                self.world.addEvent(f"Barszcz Sosnowskiego zabil {other.getNazwa()} na pozycji ({target.GetX()}, {target.GetY()})")